src/client/Microsoft.Identity.Client/AppConfig/ManagedIdentityApplicationBuilder.cs

```diff
@@ -32,107 +32,46 @@ internal ManagedIdentityApplicationBuilder(ApplicationConfiguration configuratio
+        /// For more details see https://aka.ms/msal-net-managed-identity
         /// </summary>
-        /// <param name="userAssignedId">Client ID / Resource ID of the user assigned managed identity assigned to the resource.</param>
+        /// <param name="managedIdentityConfiguration">Configuration of the Managed Identity assigned to the resource.</param>
         /// <returns>A <see cref="ManagedIdentityApplicationBuilder"/> from which to set more
         /// parameters, and to create a managed identity application instance</returns>
 #if !SUPPORTS_CONFIDENTIAL_CLIENT
         [System.ComponentModel.EditorBrowsable(System.ComponentModel.EditorBrowsableState.Never)]  // hide confidential client on mobile
 #endif
-        public static ManagedIdentityApplicationBuilder Create(string userAssignedId)
+        public static ManagedIdentityApplicationBuilder Create(ManagedIdentityConfiguration managedIdentityConfiguration)
         {
             ApplicationBase.GuardMobileFrameworks();
 
-            if (string.IsNullOrWhiteSpace(userAssignedId))
-            {
-                throw new ArgumentNullException(nameof(userAssignedId));
-            }
-
-            var config = new ApplicationConfiguration(MsalClientType.ManagedIdentityClient);
-            return new ManagedIdentityApplicationBuilder(config)
-                .WithUserAssignedManagedIdentity(userAssignedId)
-                .WithCacheSynchronization(false);
+            return new ManagedIdentityApplicationBuilder(BuildConfiguration(managedIdentityConfiguration));
         }
 
-        private ManagedIdentityApplicationBuilder WithUserAssignedManagedIdentity(string userAssignedId)
+        private static ApplicationConfiguration BuildConfiguration(ManagedIdentityConfiguration managedIdentityConfiguration)
         {
-            Config.IsUserAssignedManagedIdentity = true;
+            _ = managedIdentityConfiguration ?? throw new ArgumentNullException(nameof(managedIdentityConfiguration));
+            var config = new ApplicationConfiguration(MsalClientType.ManagedIdentityClient);
 
-            if (Guid.TryParse(userAssignedId, out _))
+            switch (managedIdentityConfiguration.IdType)
             {
-                Config.ManagedIdentityUserAssignedClientId = userAssignedId;
-            } 
-            else
-            {
-                Config.ManagedIdentityUserAssignedResourceId = userAssignedId;
+                case ManagedIdentityIdType.ClientId:
+                    config.IsUserAssignedManagedIdentity = true;
+                    config.ManagedIdentityUserAssignedClientId = managedIdentityConfiguration.UserAssignedId;
+                    break;
+                case ManagedIdentityIdType.ResourceId:
+                    config.IsUserAssignedManagedIdentity = true;
+                    config.ManagedIdentityUserAssignedResourceId = managedIdentityConfiguration.UserAssignedId;
+                    break;
```

Comment:
Can we just add a `ManagedIdentityConfiguration` property to `ApplicationConfiguration`

Reply:
Good point. We dont need all these separately in config. Removed.